pkg/packagekit/autoupdates.jsx

```diff
@@ -215,29 +215,18 @@
 
 // Returns a promise for instantiating "backend"; this will never fail, if
 // automatic updates are not supported, backend will be null.
-export function getBackend(forceReinit) {
+export function getBackend(packagekit_backend, forceReinit) {
     if (!getBackend.promise || forceReinit) {
         debug("getBackend() called first time or forceReinit passed, initializing promise");
         getBackend.promise = new Promise((resolve, reject) => {
-            cockpit.spawn(["bash", "-ec", "command -v zypper dnf apt | head -n1 | xargs --no-run-if-empty basename"], [], { err: "message" })
-                    .then(output => {
-                        output = output.trim();
-                        debug("getBackend(): detection finished, output", output);
-                        const backend = (output === "dnf") ? new DnfImpl() : undefined;
-                        // TODO: apt backend
-                        if (backend)
-                            backend.getConfig().then(() => resolve(backend));
-                        else
-                            resolve(null);
-                    })
-                    .catch(error => {
-                        // the detection shell script is supposed to always succeed
-                        console.error("automatic updates getBackend() detection failed:", error);
-                        resolve(null);
-                    });
+            const backend = (packagekit_backend === "dnf") ? new DnfImpl() : undefined;
+            // TODO: apt backend
+            if (backend)
+                backend.getConfig().then(() => resolve(backend));
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18485-20230310-145228-0edb8096-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)